src/routing_table/mod.rs

```diff
@@ -331,18 +331,31 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
         Iter { inner: self.groups.values().flat_map(iter).chain(&self.our_group) }
     }
 
+    /// Collects prefixes of all groups known by the routing table other than ours into a `HashSet`.
+    pub fn other_prefixes(&self) -> BTreeSet<Prefix<T>> {
+        self.groups.keys().cloned().collect()
+    }
+
     /// Collects prefixes of all groups known by the routing table into a `HashSet`.
     pub fn prefixes(&self) -> BTreeSet<Prefix<T>> {
         self.groups.keys().cloned().chain(iter::once(self.our_group_prefix)).collect()
     }
 
+    /// Returns all names in our section
+    // TODO: why construct a HashSet for the result?
+    pub fn our_names(&self) -> HashSet<T> {
+        let mut our_group = self.our_group.clone();
+        let _ = our_group.insert(self.our_name);
+        our_group
+    }
+
     /// If our group is the closest one to `name`, returns all names in our group *including ours*,
     /// otherwise returns `None`.
+    // TODO(refactor): we should probably replace this with self.our_names() and something checking
+    // whether a name matches our prefix.
```

Comment:
Nah, the TODO was about finding a more optimal way of doing this (not constructing a new container). But that's a more wide-spread problem; I'll just drop the TODOs for now.